### ISSUE TITLE
NVDA & Bluetooth speakers/headphones: inconsistent/insufficient volume change announcements

### ISSUE THREAD

**Issue by amirsol81** [nvaccess/nvda#15092]
In short, NVDA announces volume changes for Bluetooth headphones/speakers only when positioned on the Desktop.
### Steps to reproduce:
1. Make sure a Bluetooth speaker/headphone is connected.
2. Move to the Desktop and press the physical volume buttons on the Bluetooth speaker/headphone. Do it a couple of times.
3. Now move to anywhere other than the Desktop and do the same.
### Actual behavior:
After the volume changes, NVDA announces the new volume percentage only when positioned on the Desktop and remains silent in other apps/environments.
### Expected behavior:
Ideally - and like JAWS, NVDA should allow users to select whether they want the new volume percentage to be announced everywhere or not to be announced at all after pressing Volume Up/Down. JAWS has a setting which allows or blocks volume-related notification announcements. The current inconsistent NVDA behavior should clearly be altered.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
Alpha 28569, but this has affected all NVDA releases over the past 6 years or so.
#### Windows version:
Windows 11 Version 22H2 (OS Build 22621.1928), but it has affected Windows 8, 8.1 and 10.
#### Name and version of other software in use when reproducing the issue:
Not software, but various Bluetooth speakers and headphones tested.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Various versions backwards - 2017 and above over many years.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Haven't done so as it's not related.

**Comment by amirsol81** [comment]
@Qchristensen As mentioned privately by a user, this is not limited to Bluetooth speakers or headphones. In fact, wired speakers/headphones are equally affected if they have multimedia keys. As such, I updated both the issue title and my comments to reflect that.

**Comment by amirsol81** [comment]
@seanbudd This is a more general, non-Bluetooth issue - as reflected in the updated title/description. So it affects both Bluetooth and wired headphones/speakers.

**Comment by CyrilleB79** [comment]
This issue has already been discussed in the past.

See #10626; it had been closed by its author who has not wished to follow the request; but there other persons were interested in it (e.g. see #13197).

For the background, see also older issues such as #5726, #13415.

My conclusion is that there is no consensus on if volume notifications should be reported or not. Thus, it should be made configurable, as done in Jaws. What is sure is that people logically expect to have the same notification type (or lack of thereof), no matter if they are in Explorer, on the desktop or in any other application. Users don't care if notifications are raised by Explorer; they are even not aware of it at all.

**Comment by amirsol81** [comment]
@CyrilleB79 The point is that with newer releases of Windows 10 and all Windows 11 releases, we now have a solution for that. Like JAWS, NVDA should allow users to suppress them entirely or always hear them.

**Comment by Qchristensen** [comment]
Reported by another user today.  Exploring this with @SaschaCowley we DO announce the new volume if the focus is on file explorer, the desktop or the taskbar, but not, as noted, in other apps.  NVDA can see the information so it should be possible to report it.

As noted by my reporting user today, Narrator does report this in other apps as well.

Tested with the multimedia keys on a desktop keyboard, and the Fn+function key combinations on my laptop keyboard with the same results.

**Comment by amirsol81** [comment]
@Qchristensen Thanks for looking into this volume-related issue. I support the change as JAWS also acts like Narrator here.
